addons/survey/models/survey_survey.py

```diff
@@ -583,9 +583,32 @@ def _get_pages_or_questions(self, user_input):
             else:
                 result = self.question_and_page_ids.filtered(
                     lambda question: not question.is_page or not is_html_empty(question.description))
+                result = self._get_questions_except_invalid_conditionals(result)
 
         return result
 
+    def _get_questions_except_invalid_conditionals(self, questions):
+        """This method ignores conditional questions that are placed before their trigger"""
+        invalid_questions = self.env['survey.question']
+        for question in questions:
+            if not question.is_conditional:
+                continue
+            if (question.triggering_question_id in invalid_questions
+               or not self._is_valid_conditional_question(question)):
+                invalid_questions |= question
+        return questions - invalid_questions
+
+    def _is_valid_conditional_question(self, question):
+        trigger = question.triggering_question_id
+        if (trigger.is_page is False
+            and trigger.question_type in ['simple_choice', 'multiple_choice']
+            and trigger.suggested_answer_ids is not False
+            and (trigger.sequence < question.sequence
+                 or trigger.sequence == question.sequence
+                 and trigger.id < question.id)):
+            return True
+        return False
```

Comment:
Simplification + "cosmetics"
```suggestion
        return (
            trigger.is_page is False
            and trigger.question_type in ['simple_choice', 'multiple_choice']
            and trigger.suggested_answer_ids
            and (trigger.sequence < question.sequence
                 or trigger.sequence == question.sequence and trigger.id < question.id)
       )
```